src/mod/applications/mod_callcenter/mod_callcenter.c

```diff
@@ -2452,6 +2454,27 @@ static int agents_callback(void *pArg, int argc, char **argv, char **columnNames
 						switch_core_session_rwunlock(member_session);
 					}
 				}
+
+				if (!strcasecmp(cbt->strategy, "top-down-level-by-level")) {
+					switch_core_session_t *member_session = switch_core_session_locate(cbt->member_session_uuid);
+
+					if (member_session) {
+						char offered_agent_list[CC_OFFERED_AGENT_LIST_SIZE];
+
+						switch_channel_t *member_channel = switch_core_session_get_channel(member_session);
+
+						if ((offered_level_agent = switch_channel_get_variable(member_channel, "cc_offered_level_agent"))) {
+							snprintf(offered_agent_list, sizeof offered_agent_list, "%s,'%s'", offered_level_agent, h->agent_name);
+						} else {
+							snprintf(offered_agent_list, sizeof offered_agent_list, "'%s'", h->agent_name);
+						}
```

Comment:
@surya4 can it also clear the offered list if you are jumping to the next level. to optimize the list size.

Reply:
check if previous and new offering level is same. if not clear and add instead of appending.

Reply:
done @royalpinto